tests/driver/api.test.ts

```diff
@@ -0,0 +1,633 @@
+            const product3 = new Product({name: 'Product 3', price: 10, isCertified: true, category: 'cat 1'});
+            await Product.insertMany([product1, product2, product3]);
+            const findResp = await Product.findOne({category: 'cat 1'});
+            assert.strictEqual(findResp?.category, 'cat 1');
+        });
+        it('API ops tests Model.findOneAndDelete()', async () => {
+            const product1 = new Product({name: 'Product 1', price: 10, isCertified: true, category: 'cat 2'});
+            const product2 = new Product({name: 'Product 2', price: 10, isCertified: true, category: 'cat 2'});
+            const product3 = new Product({name: 'Product 3', price: 10, isCertified: true, category: 'cat 1'});
+            await Product.insertMany([product1, product2, product3]);
+            const deleteResp = await Product.findOneAndDelete({category: 'cat 1'});
+            assert.strictEqual(deleteResp?.category, 'cat 1');
+            //check if it exists again
+            const findDeletedDoc = await Product.findOne({category: 'cat 1'});
+            assert.strictEqual(findDeletedDoc, null);
+        });
+        it('API ops tests Model.findOneAndRemove()', async () => {
+            const product1 = new Product({name: 'Product 1', price: 10, isCertified: true, category: 'cat 2'});
+            const product2 = new Product({name: 'Product 2', price: 10, isCertified: true, category: 'cat 2'});
+            const product3 = new Product({name: 'Product 3', price: 10, isCertified: true, category: 'cat 1'});
+            await Product.insertMany([product1, product2, product3]);
+            const deleteResp = await Product.findOneAndRemove({category: 'cat 1'});
+            assert.strictEqual(deleteResp?.category, 'cat 1');
+            //check if it exists again
+            const findDeletedDoc = await Product.findOne({category: 'cat 1'});
+            assert.strictEqual(findDeletedDoc, null);
+        });
+        it('API ops tests Model.findOneAndUpdate()', async () => {
+            const product1 = new Product({name: 'Product 1', price: 10, isCertified: true, category: 'cat 2'});
+            const product2 = new Product({name: 'Product 2', price: 10, isCertified: true, category: 'cat 2'});
+            const product3 = new Product({name: 'Product 3', price: 10, isCertified: true, category: 'cat 1'});
+            await Product.insertMany([product1, product2, product3]);
+            const updateResp = await Product.findOneAndUpdate({category: 'cat 1'}, {name: 'Product 4'});
+            assert.strictEqual(updateResp?.category, 'cat 1');
+            const findOneResp = await Product.findOne({category: 'cat 1'});
+            assert.strictEqual(findOneResp?.name, 'Product 4');
+        });
+        it('API ops tests Model.hydrate()', async () => {
+            const rawProduct = {name: 'Product 1', price: 10, isCertified: true, category: 'cat 2'};
+            const product1 = Product.hydrate(rawProduct);
```

Comment:
I'd recommend not testing Model.hydrate(), that function doesn't make any database requests.

Reply:
Done, removed with a comment